Alicerce/Sources/Logging/Log.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  Alicerce.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 03/04/17.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public final class Log {
+
+    private static var providers = [LogProvider]()
+
+    public static let defaultLevel: Level = Level.error
+
+    // MARK:- Provider Management
+
+    internal static var providerCount: Int {
+        return providers.count
+    }
+
+    public class func register(_ provider: LogProvider) {
+        let matchingProviders = providers.filter { (registeredProvider) -> Bool in
+            return registeredProvider.providerInstanceId() == provider.providerInstanceId()
+        }
+
+        if matchingProviders.isEmpty {
+            providers.append(provider)
+        }
+    }
+
+    public class func unregister(_ provider: LogProvider) {
+        providers = providers.filter({ (registeredProvider) -> Bool in
+            return registeredProvider.providerInstanceId() != provider.providerInstanceId()
+        })
```

Comment:
You can simplify this into `providers.filter { $0.providerInstanceId != provider.providerInstanceId }`